addons/stock/report/report_stockpicking_operations.xml

```diff
@@ -66,100 +50,141 @@
                                         </th>
-                                        <th name="th_serial_number" class="text-center" t-if="has_serial_number">
-                                           <strong>Lot/Serial Number</strong>
-                                        </th>
-                                        <th name="th_barcode" class="text-center" t-if="has_barcode">
-                                            <strong>Product Barcode</strong>
+                                        <th name="th_barcode" class="text-center" t-if="picking_has_barcode or picking_has_serial_number">
+                                            <strong>Barcode</strong>
                                         </th>
                                     </tr>
                                 </thead>
                                 <tbody>
                                     <!-- In case you come across duplicated lines, ask NIM or LAP -->
-                                    <tr t-foreach="o.move_line_ids_without_package.sorted(lambda ml: (ml.location_id.complete_name, ml.location_dest_id.complete_name))" t-as="ml">
-                                        <td>
-                                            <span t-field="ml.product_id.display_name">Customizable Desk</span><br/>
-                                            <span t-if="ml.product_id.description_picking" t-field="ml.product_id.description_picking">Description on transfer</span>
-                                        </td>
-                                        <td>
-                                            <span t-field="ml.quantity">3.00</span>
-                                            <span t-field="ml.product_uom_id" groups="uom.group_uom">units</span>
-                                            <span t-if="ml.move_id.product_packaging_id">
-                                                <span t-if="o.state != 'done'">
-                                                    (<span t-field="ml.product_packaging_qty" t-options='{"widget": "integer"}'/> <span t-field="ml.move_id.product_packaging_id.name"/>)
+                                    <t t-foreach="o.move_ids_without_package.sorted(lambda move: (move.location_id.complete_name, move.location_dest_id.complete_name))" t-as="move">
+                                        <t t-set="move_has_serial_number" t-value="any(line.lot_id or line.lot_name for line in move.move_line_ids)" groups="stock.group_production_lot"/>
+                                        <tr>
+                                            <td>
+                                                <span t-field="move.product_id.display_name">Customizable Desk</span><br/>
+                                                <span t-if="move.description_picking != move.product_id.name and move.description_picking != move.product_id.display_name"
+                                                t-field="move.description_picking">Description on transfer</span>
```

Comment:
The same logic actually exists [here](https://github.com/odoo/odoo/blob/f6cfbfc8e01b746d304171c6f56d1f22048202a0/addons/stock/report/report_deliveryslip.xml#L74). The reason for this is that `description_picking` is set to product name by default if the description is empty. ([here](https://github.com/odoo/odoo/blob/f6cfbfc8e01b746d304171c6f56d1f22048202a0/addons/stock/models/product.py#L239))

Reply:
Yep, the logic was added by my fix :wink: 

But based on your 2nd link, it does indeed look like it's a mistake that it's currently using `move.product_id.description_picking` (because the description won't match the picking_type). I would just document that you changed it on purpose